### ISSUE TITLE
High-order derivatives 

### ISSUE THREAD

**Issue by katayooneshkofti** [lululxvi/deepxde#404]
Dear Dr. Lu 
I noticed that when we have high order derivatives (e.g. order of 4, 5, or even more) the computation time extremely increases, and unfortunately, my computer merely can handle a limited amount of points for the domain, boundary, initial conditions, and test. it seems that deepxde can learn the solution in some way (not perfectly), but when I want to modify the solution by adding the number of training and test points or even hidden layers, the computation time increases tremendously. Is there any suggestion to improve the speed of the computation?

**Comment by katayooneshkofti** [comment]
@lululxvi  Thank you, you mean the resample command?

**Comment by lululxvi** [comment]
Yes. https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.callbacks.PDEResidualResampler

**Comment by katayooneshkofti** [comment]
@lululxvi 
Thank you and I close the issue.

**Comment by katayooneshkofti** [comment]
@lululxvi 
Sorry to open this issue again. I have one more question: 
is Deepxde able to solve PDEs with high-order derivatives like d6W/dx6? (I used hessian and then gradients to obtain this order). I tried to solve a system of coupled PDEs with these high-order derivatives and I used mini-batch to help with the computations. But it seems that it takes so long to complete each iteration and the final results are not satisfactory. When I enforce hard boundaries, things get even worse. I double-checked my code, there is no error. What should I do?
Thank You

**Comment by lululxvi** [comment]
Not sure. I don't have experience on such a high order derivatives, and I haven't seen any paper working on this. This could be an interesting research topic.

**Comment by pescap** [comment]
Hi @katayooneshkofti, I am experiencing the exact same issue with hard BCs: Each successive application of the Hessian implies a 10-fold increase in computational times. Any idea @lululxvi?

Example:

```
import os
import deepxde as dde 
from deepxde.backend import tf
from functools import reduce
import numpy as np

k = 2
epochs = 10000

sigma = 1
N = 4000
N_test = 1000

geom = dde.geometry.Interval(0, 1)

def pde(x, y):
    dy_xx = dde.grad.hessian(y, x)
    
    dy1 = - dde.grad.hessian(y, x, i=0, j=0)
    for ii in range(1,k):
        dy1 =  - dde.grad.hessian(dy1, x, i=0, j=0)
    
    F_list = [x[:, 0:1] for i in range(k)]
    c = tf.exp(1/2 * k ** 2 * sigma ** 2) 
    
    A = dy1 
    C = reduce(lambda x1, x2: x1 * x2, F_list)
    return A - c * C

data = dde.data.PDE(geom, pde, [], train_distribution="pseudo", num_domain=N, solution=func, num_test=N_test)

layer_size = [1] + [50] * 4 + [1]
activation = 'tanh'
initializer = 'Glorot uniform'
net = dde.nn.FNN(layer_size, activation, initializer)

def  output_transform(x, y):
    transf_list = [x[:, 0:1] * (x[:, 0:1] - 1) for i in range(k)]
    transf = reduce(lambda x1, x2: x1 * x2, transf_list)
    return transf * y


net.apply_output_transform(output_transform)

model = dde.Model(data, net)
model.compile("adam", lr=1e-4)
losshistory, train_state = model.train(epochs=epochs)

dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

**Comment by pescap** [comment]
Similar behavior occurs with soft BCs. 
```
k = 1 -> 11s
k=2 -> 58s
k=3 -> 605s

```

Code:
```
"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch, paddle"""
import deepxde as dde
import numpy as np

k = 3
def pde(x, y):
    dy_xx = dde.grad.hessian(y, x)
    for ii in range(1,k):
        dy_xx =  - dde.grad.hessian(dy_xx, x)
    
    return dy_xx - 2

def boundary_l(x, on_boundary):
    return on_boundary and np.isclose(x[0], -1)


def boundary_r(x, on_boundary):
    return on_boundary and np.isclose(x[0], 1)


def func(x):
    return (x + 1) ** 2


geom = dde.geometry.Interval(-1, 1)
bc_l = dde.icbc.DirichletBC(geom, func, boundary_l)
bc_r = dde.icbc.RobinBC(geom, lambda X, y: y, boundary_r)
data = dde.data.PDE(geom, pde, [bc_l, bc_r], 16, 2, solution=func, num_test=100)

layer_size = [1] + [50] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001, metrics=["l2 relative error"])
losshistory, train_state = model.train(epochs=5000)

dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

**Comment by pescap** [comment]
This exponential increase is known: [Taylor-Mode Automatic Differentiation for Higher-Order Derivatives in JAX](https://openreview.net/forum?id=SkxEF3FNPH)

**Comment by pescap** [comment]
https://jax.readthedocs.io/en/latest/jax.experimental.jet.html

**Comment by pescap** [comment]
Else, one could use multi-output NN: https://arxiv.org/pdf/2103.08915.pdf

I think I will try both solutions.

**Comment by lululxvi** [comment]
Yes, we also observe the increased computational cost in https://doi.org/10.1016/j.cma.2022.114823
